airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -79,10 +80,42 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
             return True, error_msg
         return True, None
 
+    def discover(self, logger: AirbyteLogger, config) -> AirbyteCatalog:
+        """
+        Override to provide the dynamic schema generation capabilities,
+        using resource available for authenticated user.
+
+        Retrieve: Bases, Tables from each Base, generate JSON Schema for each table.
+        """
+        auth = self._auth or BigqueryAuth(config)
+        # list all bases available for authenticated account
+        for dataset in BigqueryDatasets(project_id=config["project_id"], authenticator=auth).read_records(sync_mode=SyncMode.full_refresh):
+            dataset_id = dataset.get("datasetReference")["datasetId"]
+            # list and process each table under each base to generate the JSON Schema
+            for table_info in BigqueryTables(dataset_id=dataset_id, project_id=config["project_id"], authenticator=auth).read_records(sync_mode=SyncMode.full_refresh):
+                table_id = table_info.get("tableReference")["tableId"]
+                table_obj = BigqueryTable(dataset_id=dataset_id, project_id=config["project_id"], table_id=table_id, authenticator=auth)
+
+                for table in table_obj.read_records(sync_mode=SyncMode.full_refresh):
+                    data_obj = BigqueryTableData(dataset_id=dataset_id, project_id=config["project_id"], table_id=table_id, authenticator=auth)
+                    self.streams_catalog.append(
```

Comment:
Why use a field for the class instead of a method variable? I don't see this used anywhere else.

Reply:
I did it like that because before I update this attribute in two different methods but now since I changed to only use it in one method there's no need as you said. I will update it to used just a method variable

Reply:
Which other methods? I only see usage in this method. The reason I dont like this is that now, it means this class is stateful and as a use of this class, I need to call `discover` before doing other things